internal/status/handle.go

```diff
@@ -0,0 +1,57 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/open-telemetry/opentelemetry-operator/internal/manifests"
+)
+
+const (
+	eventTypeNormal  = "Normal"
+	eventTypeWarning = "Warning"
+
+	reasonError         = "Error"
+	reasonStatusFailure = "StatusFailure"
+	reasonInfo          = "Info"
+)
+
+// HandleReconcileStatus handles updating the status of the CRDs managed by the operator.
+// TODO: make the status more useful https://github.com/open-telemetry/opentelemetry-operator/issues/1972
```

Comment:
I have a todo here because I think we should refactor this logic to make it simpler and more extendable. Will probably do after this as a solo improvement